ui/apps/platform/src/Components/PatternFly/SearchFilterChips.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { Flex, FlexItem, ChipGroup, Chip, Button, ToolbarChip } from '@patternfly/react-core';
+import noop from 'lodash/noop';
+
+import { SearchFilter } from 'types/search';
+import useURLSearch from 'hooks/useURLSearch';
+import { searchValueAsArray } from 'utils/searchUtils';
+
+import './SearchFilterChips.css';
+
+export type FilterChipGroupDescriptor = {
+    /** The name of the chip category that will be displayed in the toolbar */
+    displayName: string;
+    /** The name of the search filter category as it appears in the URL */
+    searchFilterName: string;
+    /** Optional render function for the chip. Defaults to rendering plain text inside a PatternFly `Chip` component */
+    render?: (filter: string) => React.ReactElement;
+};
+
+export type SearchFilterChipsProps = {
+    /** Callback for when the search filter changes */
+    onFilterChange?: (searchFilter: SearchFilter) => void;
+    /** The search filter categories to display */
+    filterChipGroupDescriptors: FilterChipGroupDescriptor[];
+};
+
+/**
+ * Displays and manages the search filter chips for a given set of search filter categories based
+ * on the current URL search filter
+ */
+function SearchFilterChips({
+    onFilterChange = noop,
+    filterChipGroupDescriptors,
+}: SearchFilterChipsProps) {
+    const { searchFilter, setSearchFilter } = useURLSearch();
+
+    function onChangeSearchFilter(newFilter: SearchFilter) {
+        setSearchFilter(newFilter);
+        onFilterChange(newFilter);
+    }
+
+    function onDelete(category: string, chip: ToolbarChip | string) {
+        const newSearchFilter = { ...searchFilter };
+        const newResourceFilter = searchFilter[category] as string[];
```

Comment:
Maybe my mind is slowing down too fast on a Friday. What is intended meaning of Resource?

Besides that it is a term we are trying (painfully slowly) to supersede, this seems like a re-usable component that is not limited to resources, even if that is the original use.

Reply:
Good point - resource here doesn't make sense in generic component.